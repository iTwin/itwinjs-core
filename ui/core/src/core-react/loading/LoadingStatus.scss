/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/

.loading-status-container {
  display: flex;
  align-items: center;
  opacity: 0.90;
  font-size: 12px;
}

.loading-status-message {
  flex: 1;
}

.loading-status-percent {
  flex: initial;
  margin-right: auto;
}
