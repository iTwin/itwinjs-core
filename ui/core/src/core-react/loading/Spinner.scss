/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@import "./mixins";

$spinner-size-xlarge: 96px;
$spinner-size-large:  64px;
$spinner-size-medium: 32px;
$spinner-size-small:  16px;

.core-spinner-xlarge {
  @include uicore-spinner($spinner-size-xlarge);
}

.core-spinner-large {
  @include uicore-spinner($spinner-size-large);
}

.core-spinner-medium {
  @include uicore-spinner($spinner-size-medium);
}

.core-spinner-small {
  @include uicore-spinner($spinner-size-small);
}
