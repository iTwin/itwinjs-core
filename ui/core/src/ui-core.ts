/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/

// cSpell:ignore focustrap imagecheckbox iconinput hocs numberinput numericinput

export * from "./ui-core/UiCore";

export * from "./ui-core/badge/Badge";
export * from "./ui-core/badge/BadgeUtilities";
export * from "./ui-core/badge/BetaBadge";
export * from "./ui-core/badge/NewBadge";

export * from "./ui-core/base/Div";
export * from "./ui-core/base/DivWithOutsideClick";
export * from "./ui-core/base/Centered";
export * from "./ui-core/base/FillCentered";
export * from "./ui-core/base/FlexWrapContainer";
export * from "./ui-core/base/PointerEvents";
export * from "./ui-core/base/ScrollView";

export * from "./ui-core/button/Button";
export * from "./ui-core/button/UnderlinedButton";

export * from "./ui-core/checklistbox/CheckListBox";

export * from "./ui-core/contextmenu/ContextMenu";
export * from "./ui-core/contextmenu/ContextMenuDirection";
export * from "./ui-core/contextmenu/ContextMenuDivider";
export * from "./ui-core/contextmenu/ContextMenuItem";
export * from "./ui-core/contextmenu/ContextSubMenu";
export * from "./ui-core/contextmenu/GlobalContextMenu";
export * from "./ui-core/contextmenu/PopupContextMenu";

export * from "./ui-core/cube/Cube";

export * from "./ui-core/dialog/Dialog";

export * from "./ui-core/elementseparator/ElementSeparator";

export * from "./ui-core/enums/Alignment";
export * from "./ui-core/enums/CheckBoxState";
export * from "./ui-core/enums/Orientation";
export * from "./ui-core/enums/SortDirection";
export * from "./ui-core/enums/TimeFormat";

export * from "./ui-core/expandable/ExpandableList";
export * from "./ui-core/expandable/ExpandableBlock";

export * from "./ui-core/focus/ItemKeyboardNavigator";
export * from "./ui-core/focustrap/FocusTrap";

export * from "./ui-core/form/Field";
export * from "./ui-core/form/Form";

export * from "./ui-core/hocs/withIsPressed";
export * from "./ui-core/hocs/withOnOutsideClick";
export * from "./ui-core/hocs/withTimeout";

export * from "./ui-core/icons/IconComponent";
export * from "./ui-core/icons/SvgPath";
export * from "./ui-core/icons/SvgSprite";
export * from "./ui-core/icons/WebFontIcon";

export * from "./ui-core/autosuggest/AutoSuggest";

export * from "./ui-core/checkbox/Checkbox";

export * from "./ui-core/imagecheckbox/ImageCheckBox";

export * from "./ui-core/inputs/Input";
export * from "./ui-core/inputs/InputLabel";
export * from "./ui-core/inputs/InputStatus";
export * from "./ui-core/inputs/iconinput/IconInput";
export * from "./ui-core/inputs/LabeledComponentProps";
export * from "./ui-core/inputs/LabeledInput";
export * from "./ui-core/inputs/LabeledTextarea";
export * from "./ui-core/inputs/numberinput/NumberInput";
export * from "./ui-core/inputs/numericinput/NumericInput";
export * from "./ui-core/inputs/numericinput/ReactNumericInput";
export * from "./ui-core/inputs/Textarea";

export * from "./ui-core/listbox/Listbox";

export * from "./ui-core/loading/LoadingBar";
export * from "./ui-core/loading/LoadingPrompt";
export * from "./ui-core/loading/LoadingSpinner";
export * from "./ui-core/loading/LoadingStatus";
export * from "./ui-core/loading/Spinner";

export * from "./ui-core/messagebox/MessageBox";

export * from "./ui-core/notification/MessageRenderer";
export * from "./ui-core/notification/MessageType";

export * from "./ui-core/popup/Popup";

export * from "./ui-core/progress-indicators/ProgressBar";
export * from "./ui-core/progress-indicators/ProgressSpinner";

export * from "./ui-core/radialmenu/RadialMenu";
export * from "./ui-core/radialmenu/Annulus";
export * from "./ui-core/radio/Radio";

export * from "./ui-core/select/LabeledSelect";
export * from "./ui-core/select/Select";
export * from "./ui-core/select/ThemedSelect";
export * from "./ui-core/select/LabeledThemedSelect";
export * from "./ui-core/searchbox/SearchBox";

export * from "./ui-core/settings/SettingsManager";
export * from "./ui-core/settings/SettingsContainer";

export * from "./ui-core/slider/Slider";

export * from "./ui-core/splitbutton/SplitButton";

export * from "./ui-core/tabs/HorizontalTabs";
export * from "./ui-core/tabs/VerticalTabs";
export * from "./ui-core/tabs/Tabs";

export * from "./ui-core/text/BodyText";
export * from "./ui-core/text/BlockText";
export * from "./ui-core/text/DisabledText";
export * from "./ui-core/text/FilteredText";
export * from "./ui-core/text/Headline";
export * from "./ui-core/text/LeadingText";
export * from "./ui-core/text/LeadingText2";
export * from "./ui-core/text/MutedText";
export * from "./ui-core/text/SmallText";
export * from "./ui-core/text/Subheading";
export * from "./ui-core/text/Subheading2";
export * from "./ui-core/text/StyledText";
export * from "./ui-core/text/TextProps";
export * from "./ui-core/text/Title";
export * from "./ui-core/text/Title2";

export * from "./ui-core/tiles/FeaturedTile";
export * from "./ui-core/tiles/MinimalFeaturedTile";
export * from "./ui-core/tiles/MinimalTile";
export * from "./ui-core/tiles/Tile";

export * from "./ui-core/toggle/Toggle";
export * from "./ui-core/toggle/LabeledToggle";

export * from "./ui-core/tooltip/Tooltip";

export { ExpansionToggle, ExpansionToggleProps } from "./ui-core/tree/ExpansionToggle";
export { TreeBranch, TreeBranchProps } from "./ui-core/tree/Branch";
export { TreeNode, TreeNodeProps, NodeCheckboxProps, NodeCheckboxRenderer, NodeCheckboxRenderProps } from "./ui-core/tree/Node";
export { Tree, TreeProps } from "./ui-core/tree/Tree";
export { TreeNodePlaceholder, TreeNodePlaceholderProps } from "./ui-core/tree/Placeholder";

export * from "./ui-core/uisettings/UiSetting";
export * from "./ui-core/uisettings/UiSettings";
export * from "./ui-core/uisettings/LocalUiSettings";
export * from "./ui-core/uisettings/SessionUiSettings";

export * from "./ui-core/utils/IconHelper";
export * from "./ui-core/utils/Point";
export * from "./ui-core/utils/Props";
export * from "./ui-core/utils/Rectangle";
export * from "./ui-core/utils/Size";
export * from "./ui-core/utils/Timer";
export * from "./ui-core/utils/UiEvent";
export * from "./ui-core/utils/flattenChildren";
export * from "./ui-core/utils/getBestBWContrastColor";
export * from "./ui-core/utils/getCssVariable";
export * from "./ui-core/utils/getDisplayName";
export * from "./ui-core/utils/getUserColor";
export * from "./ui-core/utils/shallowDiffers";
export * from "./ui-core/utils/typeUtils";
export * from "./ui-core/utils/isPromiseLike";
export * from "./ui-core/utils/ScrollPositionMaintainer";
export * from "./ui-core/utils/hooks/useDisposable";
export * from "./ui-core/utils/hooks/useEffectSkipFirst";
export * from "./ui-core/utils/hooks/ResizeObserverPolyfill";
export * from "./ui-core/utils/hooks/useOnOutsideClick";
export * from "./ui-core/utils/hooks/useProximityToMouse";
export * from "./ui-core/utils/hooks/useRefEffect";
export * from "./ui-core/utils/hooks/useRefs";
export * from "./ui-core/utils/hooks/useRefState";
export * from "./ui-core/utils/hooks/useResizeObserver";
export * from "./ui-core/utils/hooks/useTargeted";
export * from "./ui-core/utils/hooks/useWidgetOpacityContext";

/** @docs-package-description
 * The ui-core package contains general purpose React components, such as Dialog, MessageBox, SearchBox, RadialMenu and SplitButton.
 * For more information, see [learning about ui-core]($docs/learning/ui/core/index.md).
 */
/**
 * @docs-group-description AutoSuggest
 * Component for input with an auto-suggestion dropdown.
 */
/**
 * @docs-group-description Base
 * Low-level classes and components for building application UI.
 */
/**
 * @docs-group-description Button
 * Components for working with various Buttons.
 */
/**
 * @docs-group-description Checkbox
 * Component is a wrapper for the `<input type="checkbox">` HTML element.
 */
/**
 * @docs-group-description CheckListBox
 * Components for working with a Check listbox.
 */
/**
 * @docs-group-description Common
 * Common classes and enums used across various UI components.
 */
/**
 * @docs-group-description ContextMenu
 * Components for working with a Context Menu.
 */
/**
 * @docs-group-description Cube
 * Component for 3D Cube.
 */
/**
 * @docs-group-description Dialog
 * Components for working with a Dialog or MessageBox.
 */
/**
 * @docs-group-description ElementSeparator
 * Components for working with a ElementSeparator.
 */
/**
 * @docs-group-description Expandable
 * Components for working with a ExpandableBlock or ExpandableList.
 */
/**
 * @docs-group-description Form
 * Components used to create a Form using supplied properties to specify fields.
 */
/**
 * @docs-group-description Icon
 * Component that renders ui-core icon when given an icon name or SVG source.
 */
/**
 * @docs-group-description Inputs
 * Components for working with input controls, such as Input, IconInput, NumberInput and Textarea.
 */
/**
 * @docs-group-description Loading
 * Components for working with Loading spinners and progress indicators and bars.
 */
/**
 * @docs-group-description Notification
 * Components for working with messages and tooltips.
 */
/**
 * @docs-group-description Popup
 * Components for working with a Popup.
 */
/**
 * @docs-group-description RadialMenu
 * Components for working with a RadialMenu.
 */
/**
 * @docs-group-description Radio
 * Component is a wrapper for the `<input type="radio">` HTML element.
 */
/**
 * @docs-group-description SearchBox
 * Components for working with a SearchBox.
 */
/**
 * @docs-group-description Select
 * Component is a wrapper for the `<select>` HTML element.
 */
/**
 * @docs-group-description Settings
 * Manager and UI Components that allow users to modify settings for different packages and extensions.
 */
/**
 * @docs-group-description Slider
 * Component displays a range slider with thumbs for changing the value.
 */
/**
 * @docs-group-description SplitButton
 * Components for working with a SplitButton.
 */
/**
 * @docs-group-description Tabs
 * Components for working with horizontal or vertical tabs.
 */
/**
 * @docs-group-description Text
 * Components for working with styled text.
 */
/**
 * @docs-group-description Tiles
 * Components for a container rendering elements that can be grouped together.
 */
/**
 * @docs-group-description Toggle
 * Components for working with a Toggle switch.
 */
/**
 * @docs-group-description Tooltip
 * Components for working with a Tooltip.
 */
/**
 * @docs-group-description Tree
 * Presentation React components for working with a Tree.
 */
/**
 * @docs-group-description UiSettings
 * Interfaces and classes for working with persistent UI settings.
 */
/**
 * @docs-group-description Utilities
 * Various utility classes, functions and React hooks for working with a UI.
 */
