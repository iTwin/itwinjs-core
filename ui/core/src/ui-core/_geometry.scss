/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@mixin triangle-common {
  width: 0;
  height: 0;
  border-style: solid;
}

@mixin triangle-left-color($color) {
  border-color: transparent $color transparent transparent;
}

@mixin triangle-left($height, $width: $height, $color: currentColor) {
  border-width: ($height / 2) $width ($height / 2) 0;

  @include triangle-common;
  @include triangle-left-color($color);
}

@mixin triangle-top-color($color) {
  border-color: transparent transparent $color transparent;
}

@mixin triangle-top($height, $width: $height, $color: currentColor) {
  border-width: 0 ($width / 2) $height ($width / 2);

  @include triangle-common;
  @include triangle-top-color($color);
}

@mixin triangle-right-color($color) {
  border-color: transparent transparent transparent $color;
}

@mixin triangle-right($height, $width: $height, $color: currentColor) {
  border-width: ($height / 2) 0 ($height / 2) $width;

  @include triangle-common;
  @include triangle-right-color($color);
}

@mixin triangle-bottom-color($color) {
  border-color: $color transparent transparent transparent;
}

@mixin triangle-bottom($height, $width: $height, $color: currentColor) {
  border-width: $height ($width / 2) 0 ($width / 2);

  @include triangle-common;
  @include triangle-bottom-color($color);
}

@mixin triangle-bottom-right-color($color) {
  border-color: transparent transparent $color transparent;
}

@mixin triangle-bottom-right($height, $width: $height, $color: currentColor) {
  border-width: 0 0 $height $width;

  @include triangle-common;
  @include triangle-bottom-right-color($color);
}

@mixin oval($width, $height: $width) {
  width: $width;
  height: $height;
  border-radius: #{$width / 2} / #{$height / 2};
}
