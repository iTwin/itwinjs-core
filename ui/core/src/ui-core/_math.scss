/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@function sqrt($x, $iterations: 10) {
  $sqr: 1;

  @for $i from 1 through $iterations {
    $sqr: ($sqr - ($sqr * $sqr - $x) / (2 * $sqr));
  }

  @return $sqr;
}
