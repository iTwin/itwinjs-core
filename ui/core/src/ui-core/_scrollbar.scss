/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@import "./style/themecolors.scss";

$uicore-scrollbar-width: 12px;
$uicore-scrollbar-padding: 2px;

@mixin uicore-scrollbar($color: $buic-background-scrollbar, $width: $uicore-scrollbar-width, $padding: $uicore-scrollbar-padding) {
  &::-webkit-scrollbar {
    width: $width;
    height: $width;
  }

  &::-webkit-scrollbar-thumb {
    border-radius: $width;
    background-color: $color;
    background-clip: content-box;
    border-style: solid;
    border-color: transparent;
    border-width: $padding;

    &:hover {
      background-color: $buic-background-scrollbar-hover;
    }
  }

  &::-webkit-scrollbar-corner {
    background-color: transparent;
  }

  &::-webkit-scrollbar-track-piece:corner-present {
    margin-bottom: -2 * $padding;
    margin-right: -2 * $padding;
  }
}

@mixin uicore-hidden-scrollbar {
  &::-webkit-scrollbar {
    width: 0;
    height: 0;
  }
}

@mixin uicore-touch-scrolling {
  @supports (-webkit-overflow-scrolling: touch) {
    overflow-y: scroll; /* overflow-y has to be scroll, not auto */
    -webkit-overflow-scrolling: touch;
  }
}
