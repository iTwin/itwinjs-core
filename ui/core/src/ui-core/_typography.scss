/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@import "./style/opacity.scss";
@import "./style/typography.scss";

$title-weight: $uicore-font-weight-normal;
$title-size: $uicore-font-size-leading;
$title-opacity: $uicore-opacity-1;

$body-weight: $uicore-font-weight-light;
$body-size: $uicore-font-size;
$body-opacity: $uicore-opacity-2;

$body1-weight: $uicore-font-weight-normal;
$body1-size: $uicore-font-size;
$body1-opacity: $uicore-opacity-2;

@mixin title($rgb: unset) {
  font-weight: $title-weight;
  font-size: $title-size;

  @if $rgb != unset {
    color: rgba($rgb, $title-opacity);
  }
}

@mixin body($rgb: unset) {
  font-weight: $body-weight;
  font-size: $body-size;

  @if $rgb != unset {
    color: rgba($rgb, $body-opacity);
  }
}

@mixin body1($rgb: unset) {
  font-weight: $body1-weight;
  font-size: $body1-size;

  @if $rgb != unset {
    color: rgba($rgb, $body1-opacity);
  }
}