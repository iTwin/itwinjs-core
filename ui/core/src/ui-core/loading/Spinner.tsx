/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
/** @packageDocumentation
 * @module Loading
 */

import "./Spinner.scss";
import * as React from "react";

// cspell:ignore xlarge

/** Size for [[Spinner]] component
 * @public
 */
export enum SpinnerSize {
  /** width/height of 16px */
  Small,
  /** width/height of 32px */
  Medium,
  /** width/height of 64px */
  Large,
  /** width/height of 96px */
  XLarge,
}

/** Properties for [[Spinner]] component
 * @public
 */
export interface SpinnerProps {
  /** Size of spinner */
  size?: SpinnerSize;
  /** CSS class for size of spinner */
  sizeClass?: string;
}

/**
 * An indeterminate spinner component.
 * @public
 */
export class Spinner extends React.PureComponent<SpinnerProps> {

  public render() {
    let sizeClass: string;

    if (this.props.sizeClass)
      sizeClass = this.props.sizeClass;
    else {
      switch (this.props.size) {
        case SpinnerSize.Small:
          sizeClass = "core-spinner-small";
          break;
        case SpinnerSize.Large:
          sizeClass = "core-spinner-large";
          break;
        case SpinnerSize.XLarge:
          sizeClass = "core-spinner-xlarge";
          break;
        case SpinnerSize.Medium:
        default:
          sizeClass = "core-spinner-medium";
          break;
      }
    }

    return (
      <div className={sizeClass} />
    );
  }
}
