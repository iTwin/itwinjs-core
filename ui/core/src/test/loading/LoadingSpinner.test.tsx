/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
import { mount, shallow } from "enzyme";
import * as React from "react";
import { LoadingSpinner } from "../../ui-core";
import { SpinnerSize } from "../../ui-core/loading/Spinner";

/* eslint-disable deprecation/deprecation */

describe("<LoadingSpinner />", () => {
  it("should render", () => {
    const wrapper = mount(
      <LoadingSpinner />,
    );
    wrapper.unmount();
  });

  it("renders correctly", () => {
    shallow(
      <LoadingSpinner />,
    ).should.matchSnapshot();
  });

  it("renders with message correctly", () => {
    shallow(<LoadingSpinner message="test" />).should.matchSnapshot();
  });

  it("renders with message and position correctly", () => {
    shallow(<LoadingSpinner message="test" messageOnTop={true} />).should.matchSnapshot();
  });

  // Tests for Deprecated SpinnerSize
  it("renders with Small size correctly", () => {
    shallow(<LoadingSpinner size={SpinnerSize.Small} />).should.matchSnapshot();
  });
  it("renders with Medium size correctly", () => {
    shallow(<LoadingSpinner size={SpinnerSize.Medium} />).should.matchSnapshot();
  });
  it("renders with Large size correctly", () => {
    shallow(<LoadingSpinner size={SpinnerSize.Large} />).should.matchSnapshot();
  });
  it("renders with XLarge size correctly", () => {
    shallow(<LoadingSpinner size={SpinnerSize.XLarge} />).should.matchSnapshot();
  });

});
