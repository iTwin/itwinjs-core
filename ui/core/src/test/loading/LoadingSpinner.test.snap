// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingSpinner /> renders correctly 1`] = `
<div
  className="core-ls"
>
  <ProgressRadial
    indeterminate={true}
  />
</div>
`;

exports[`<LoadingSpinner /> renders with Large size correctly 1`] = `
<div
  className="core-ls"
>
  <ProgressRadial
    indeterminate={true}
    size="large"
  />
</div>
`;

exports[`<LoadingSpinner /> renders with Medium size correctly 1`] = `
<div
  className="core-ls"
>
  <ProgressRadial
    indeterminate={true}
  />
</div>
`;

exports[`<LoadingSpinner /> renders with Small size correctly 1`] = `
<div
  className="core-ls"
>
  <ProgressRadial
    indeterminate={true}
    size="x-small"
  />
</div>
`;

exports[`<LoadingSpinner /> renders with XLarge size correctly 1`] = `
<div
  className="core-ls"
>
  <ProgressRadial
    indeterminate={true}
    size="large"
  />
</div>
`;

exports[`<LoadingSpinner /> renders with message and position correctly 1`] = `
<div
  className="core-ls"
>
  <span
    className="ls-message-top"
  >
    test
  </span>
  <ProgressRadial
    indeterminate={true}
  />
</div>
`;

exports[`<LoadingSpinner /> renders with message correctly 1`] = `
<div
  className="core-ls"
>
  <ProgressRadial
    indeterminate={true}
  />
  <span
    className="ls-message-bottom"
  >
    test
  </span>
</div>
`;
